apps/web/src/components/evaluations/rule/SemanticSimilarity.tsx

```diff
@@ -0,0 +1,159 @@
+  ResultRowHeadersProps,
+} from '../index'
+
+const specification = RuleEvaluationSemanticSimilaritySpecification
+export default {
+  ...specification,
+  icon: 'equalApproximately' as IconName,
+  ConfigurationForm: ConfigurationForm,
+  ResultBadge: ResultBadge,
+  ResultRowHeaders: ResultRowHeaders,
+  ResultRowCells: ResultRowCells,
+  resultPanelTabs: [],
+  ResultPanelMetadata: ResultPanelMetadata,
+  ResultPanelContent: ResultPanelContent,
+  chartConfiguration: chartConfiguration,
+}
+
+const ALGORITHM_OPTIONS =
+  specification.configuration.shape.algorithm.options.map((option) => ({
+    label: option.toUpperCase().split('_').join(' '),
+    value: option,
+  }))
+
+function ConfigurationForm({
+  configuration,
+  setConfiguration,
+  disabled,
+}: ConfigurationFormProps<
+  EvaluationType.Rule,
+  RuleEvaluationMetric.SemanticSimilarity
+>) {
+  return (
+    <>
+      <Select
+        value={configuration.algorithm ?? ''}
+        name='algorithm'
+        label='Algorithm'
+        description='How to measure percentage of similarity'
+        placeholder='Select an algorithm'
+        options={ALGORITHM_OPTIONS}
+        onChange={(value) =>
+          setConfiguration({ ...configuration, algorithm: value })
+        }
+        disabled={disabled}
+        required
+      />
+      <NumberInput
+        value={configuration.minSimilarity ?? undefined}
+        name='minSimilarity'
+        label='Minimum similarity'
+        description='The minimum percentage of similarity of the response'
+        placeholder='No minimum'
+        min={0}
+        max={100}
+        onChange={(value) =>
+          setConfiguration({ ...configuration, minSimilarity: value })
+        }
+        className='w-full'
+        disabled={disabled}
+        required
+      />
+      <NumberInput
+        value={configuration.maxSimilarity ?? undefined}
+        name='maxSimilarity'
+        label='Maximum similarity'
+        description='The maximum percentage of similarity of the response'
+        placeholder='No maximum'
+        min={0}
+        max={100}
+        onChange={(value) =>
+          setConfiguration({ ...configuration, maxSimilarity: value })
+        }
+        className='w-full'
+        disabled={disabled}
+        required
+      />
+    </>
+  )
+}
+
+function ResultBadge({
+  result,
+}: ResultBadgeProps<
+  EvaluationType.Rule,
+  RuleEvaluationMetric.SemanticSimilarity
+>) {
+  return <>{result.score!.toFixed(0)}% similar</>
+}
+
+function ResultRowHeaders(
+  _props: ResultRowHeadersProps<
+    EvaluationType.Rule,
+    RuleEvaluationMetric.SemanticSimilarity
+  >,
+) {
+  return <></>
+}
+
+function ResultRowCells(
+  _props: ResultRowCellsProps<
+    EvaluationType.Rule,
+    RuleEvaluationMetric.SemanticSimilarity
+  >,
+) {
+  return <></>
+}
+
+function ResultPanelMetadata(
+  _props: ResultPanelProps<
+    EvaluationType.Rule,
+    RuleEvaluationMetric.SemanticSimilarity
+  >,
+) {
+  return <></>
+}
```

Comment:
Instead of this I would do a fallback in the place where these components are used. And in the spec for this one set it to `null`